api4/api.go

```diff
@@ -51,11 +51,12 @@ type Routes struct {
 	ChannelModerations       *mux.Router // 'api/v4/channels/{channel_id:[A-Za-z0-9]+}/moderations'
 	ChannelCategories        *mux.Router // 'api/v4/users/{user_id:[A-Za-z0-9]+}/teams/{team_id:[A-Za-z0-9]+}/channels/categories'
 
-	Posts           *mux.Router // 'api/v4/posts'
-	Post            *mux.Router // 'api/v4/posts/{post_id:[A-Za-z0-9]+}'
-	PostsForChannel *mux.Router // 'api/v4/channels/{channel_id:[A-Za-z0-9]+}/posts'
-	PostsForUser    *mux.Router // 'api/v4/users/{user_id:[A-Za-z0-9]+}/posts'
-	PostForUser     *mux.Router // 'api/v4/users/{user_id:[A-Za-z0-9]+}/posts/{post_id:[A-Za-z0-9]+}'
+	Posts                *mux.Router // 'api/v4/posts'
+	Post                 *mux.Router // 'api/v4/posts/{post_id:[A-Za-z0-9]+}'
+	PostsForChannel      *mux.Router // 'api/v4/channels/{channel_id:[A-Za-z0-9]+}/posts'
+	PostsForUser         *mux.Router // 'api/v4/users/{user_id:[A-Za-z0-9]+}/posts'
+	PostForUser          *mux.Router // 'api/v4/users/{user_id:[A-Za-z0-9]+}/posts/{post_id:[A-Za-z0-9]+}'
+	PostAcknowledgements *mux.Router // 'api/v4/posts/{post_id:[A-Za-z0-9]+}/ack'
```

Comment:
nit: I'm surprised to see a route at this level that's technically a descendant of the existing `Post` route. (In that I conceptually think of the `Post` route as defining everything underneath that path.)

I don't think there's anything strictly wrong with this, but just calling it out.

Reply:
Ooh, yes! The best suggestion so far :)